python/ray/data/datasource/parquet_datasource.py

```diff
@@ -419,31 +419,31 @@ def _fetch_metadata(
 
 
 def _sample_piece(
+    reader_args,
+    columns,
+    schema,
     file_piece: _SerializedPiece,
-    row_group_id: int,
 ) -> float:
-    # Sample the `row_group_id`-th row group from file piece `serialized_piece`.
+    # Sample the first rows batch from file piece `serialized_piece`.
     # Return the encoding ratio calculated from the sampled rows.
     piece = _deserialize_pieces_with_retry([file_piece])[0]
 
-    # If required row group index is out of boundary, sample the last row group.
-    row_group_id = min(piece.num_row_groups - 1, row_group_id)
-    assert (
-        row_group_id >= 0 and row_group_id <= piece.num_row_groups - 1
-    ), f"Required row group id {row_group_id} is not in expected bound"
-
-    row_group = piece.subset(row_group_ids=[row_group_id])
-    metadata = row_group.metadata.row_group(0)
-    num_rows = min(PARQUET_ENCODING_RATIO_ESTIMATE_NUM_ROWS, metadata.num_rows)
-    assert num_rows > 0 and metadata.num_rows > 0, (
-        f"Sampled number of rows: {num_rows} and total number of rows: "
-        f"{metadata.num_rows} should be positive"
+    batches = piece.to_batches(
+        columns=columns,
+        schema=schema,
+        batch_size=PARQUET_READER_ROW_BATCH_SIZE,
```

Comment:
I'm a bit concerned that we're going to be reading a lot more data now, going from 5 rows to 100k rows, which could be a lot slower/heavier for very wide tables.

Reply:
Yeah I shared the same concern. Read 2GB takes roughly 14 seconds, is the baseline we have now, which is not too bad. Let me explore more if I can find anything better.